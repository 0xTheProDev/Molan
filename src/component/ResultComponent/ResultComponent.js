// @flow
import React, { PureComponent } from 'react';
import { Grid, Form, TextArea } from 'semantic-ui-react';

import LoaderComponent from 'component/LoaderComponent';

type Result = {
  input: string,
  output: string,
  status: string,
};

type ResultComponentProps = {
  result: Result,
  submit: boolean,
  onSubmit: (submitting: boolean) => void,
};

export class ResultComponent extends PureComponent<ResultComponentProps> {
  result: Result & {
    color: string,
  };

  constructor(props) {
    super(props);
    this.result = {};
  }

  componentWillReceiveProps(props: ResultComponentProps) {
    const { result } = props;
    const { onSubmit } = this.props;
    const { getColor } = this;

    if (!this.result.hasOwnProperty('id') || this.result.id !== result.id) {
      this.result = result;
      this.result.color = getColor(result.status)
      onSubmit(false);
    }
  }

  getColor = (status: string) => {
    switch (status) {
      case 'Success': return 'green';
      case 'Runtime Error': return 'red';
      case 'Compile Error': return 'yellow';
      default: return 'teal';
    }
  };

  render() {
    const { submit } = this.props;
    const { result } = this;

    if (submit === true) {
      return (<LoaderComponent />);
    }

    return (
      <Grid>
        <Grid.Row>
          <Grid.Column color={result.color}>Status:&emsp;{result.status}</Grid.Column>
        </Grid.Row>
        <Grid.Row columns={1}>
          <Grid.Column stretched>
            <Form>
              <label htmlFor="custom_input">Custom Input:</label>
              <TextArea
                disabled
                autoHeight
                value={result.input}
                placeholder="No Input to show"
                id="custom_input"
              />
            </Form>
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={1}>
          <Grid.Column stretched>
            <Form>
              <label htmlFor="custom_output">Custom Output:</label>
              <TextArea
                disabled
                autoHeight
                value={result.output}
                placeholder="No Output to show"
                id="custom_output"
              />
            </Form>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}
