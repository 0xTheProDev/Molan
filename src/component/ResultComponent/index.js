import React, { Component } from 'react';
import { Grid, Form, TextArea } from 'semantic-ui-react';

export default class ResultComponent extends Component {
    render() {
        return (
            <Grid>
                <Grid.Row>
                    <Grid.Column>Status</Grid.Column>
                </Grid.Row>
                <Grid.Row columns={1}>
                    <Grid.Column stretched>
                        <Form>
                            <label htmlFor='custom_input'>Custom Input:</label>
                            <TextArea
                              disabled
                              autoHeight
                              placeholder='No Input to show'
                              id='custom_input'
                            />
                        </Form>
                    </Grid.Column>
                </Grid.Row>
                <Grid.Row columns={1}>
                    <Grid.Column stretched>
                        <Form>
                            <label htmlFor='custom_output'>Custom Output:</label>
                            <TextArea
                              readOnly
                              autoHeight
                              placeholder='No Output to show'
                              id='custom_output'
                            />
                        </Form>
                    </Grid.Column>
                </Grid.Row>
            </Grid>
        );
    }
}