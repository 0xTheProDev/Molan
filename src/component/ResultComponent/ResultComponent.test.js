import React from 'react';

import { ResultComponent } from './ResultComponent';
import { Item } from 'semantic-ui-react';

describe('Result Component', () => {
  let
    onSubmitStub,
    minProps;

  beforeAll(() => {
    onSubmitStub = jest.fn();
    minProps = {
      result: null,
      submit: false,
      onSubmit: onSubmitStub,
    };
  });

  it('should render', () => {
    const wrapper = shallow(<ResultComponent {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Loader if results not available', () => {
    const diffProps = {
      submit: true,
    };
    const wrapper = shallow(<ResultComponent {...minProps} {...diffProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render results if available', () => {
    const diffProps = {
      result: {
        status: 'Success',
        input: '0',
        output: 'Zero',
      },
    };
    const wrapper = shallow(<ResultComponent {...minProps} {...diffProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
