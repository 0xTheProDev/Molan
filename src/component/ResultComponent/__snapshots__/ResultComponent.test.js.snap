// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component should render 1`] = `
<Grid>
  <GridRow>
    <GridColumn>
      Status: 
    </GridColumn>
  </GridRow>
  <GridRow
    columns={1}
  >
    <GridColumn
      stretched={true}
    >
      <Form
        as="form"
      >
        <label
          htmlFor="custom_input"
        >
          Custom Input:
        </label>
        <TextArea
          as="textarea"
          autoHeight={true}
          disabled={true}
          id="custom_input"
          placeholder="No Input to show"
          rows={3}
        />
      </Form>
    </GridColumn>
  </GridRow>
  <GridRow
    columns={1}
  >
    <GridColumn
      stretched={true}
    >
      <Form
        as="form"
      >
        <label
          htmlFor="custom_output"
        >
          Custom Output:
        </label>
        <TextArea
          as="textarea"
          autoHeight={true}
          disabled={true}
          id="custom_output"
          placeholder="No Output to show"
          rows={3}
        />
      </Form>
    </GridColumn>
  </GridRow>
</Grid>
`;

exports[`Result Component should render Loader if results not available 1`] = `<LoaderComponent />`;

exports[`Result Component should render results if available 1`] = `
<Grid>
  <GridRow>
    <GridColumn>
      Status: 
    </GridColumn>
  </GridRow>
  <GridRow
    columns={1}
  >
    <GridColumn
      stretched={true}
    >
      <Form
        as="form"
      >
        <label
          htmlFor="custom_input"
        >
          Custom Input:
        </label>
        <TextArea
          as="textarea"
          autoHeight={true}
          disabled={true}
          id="custom_input"
          placeholder="No Input to show"
          rows={3}
        />
      </Form>
    </GridColumn>
  </GridRow>
  <GridRow
    columns={1}
  >
    <GridColumn
      stretched={true}
    >
      <Form
        as="form"
      >
        <label
          htmlFor="custom_output"
        >
          Custom Output:
        </label>
        <TextArea
          as="textarea"
          autoHeight={true}
          disabled={true}
          id="custom_output"
          placeholder="No Output to show"
          rows={3}
        />
      </Form>
    </GridColumn>
  </GridRow>
</Grid>
`;
